pkg/azurestore/azureservice.go

```diff
@@ -67,87 +72,92 @@ type AzBlob interface {
 	case "blob":
-		containerAccessType = azblob.PublicAccessBlob
-	case "":
+		containerAccessType = container.PublicAccessTypeBlob
 	default:
-		containerAccessType = azblob.PublicAccessNone
+		containerAccessType = ""
+	}
+
+	// Do not care about response since it will fail if container exists and create if it does not.
+	_, err = containerClient.Create(context.Background(), &container.CreateOptions{
+		Access: to.Ptr(containerAccessType),
+	})
+	if err != nil && !strings.Contains(err.Error(), "ContainerAlreadyExists") {
+		return nil, err
 	}
 
-	// Does not support the premium access tiers
-	var blobAccessTierType azblob.AccessTierType
+	var blobAccessTier *blob.AccessTier
 	switch config.BlobAccessTier {
 	case "archive":
-		blobAccessTierType = azblob.AccessTierArchive
+		blobAccessTier = to.Ptr(blob.AccessTierArchive)
 	case "cool":
-		blobAccessTierType = azblob.AccessTierCool
+		blobAccessTier = to.Ptr(blob.AccessTierCool)
 	case "hot":
-		blobAccessTierType = azblob.AccessTierHot
+		blobAccessTier = to.Ptr(blob.AccessTierHot)
 	case "":
 	default:
-		blobAccessTierType = azblob.DefaultAccessTier
+		blobAccessTier = nil
 	}
 
-	// The pipeline specifies things like retry policies, logging, deserialization of HTTP response payloads, and more.
-	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
-	cURL, _ := url.Parse(fmt.Sprintf("%s/%s", config.Endpoint, config.ContainerName))
-
-	// Get the ContainerURL URL
-	containerURL := azblob.NewContainerURL(*cURL, p)
-	// Do not care about response since it will fail if container exists and create if it does not.
-	_, _ = containerURL.Create(context.Background(), azblob.Metadata{}, containerAccessType)
-
 	return &azService{
-		BlobAccessTier: blobAccessTierType,
-		ContainerURL:   &containerURL,
-		ContainerName:  config.ContainerName,
+		ContainerClient: containerClient,
+		ContainerName:   config.ContainerName,
+		BlobAccessTier:  blobAccessTier,
 	}, nil
 }
 
 // Determine if we return a InfoBlob or BlockBlob, based on the name
 func (service *azService) NewBlob(ctx context.Context, name string) (AzBlob, error) {
-	var fileBlob AzBlob
-	bb := service.ContainerURL.NewBlockBlobURL(name)
+	blobClient := service.ContainerClient.NewBlockBlobClient(name)
 	if strings.HasSuffix(name, InfoBlobSuffix) {
-		fileBlob = &InfoBlob{
-			Blob: &bb,
-		}
-	} else {
-		fileBlob = &BlockBlob{
-			Blob:       &bb,
-			Indexes:    []int{},
-			AccessTier: service.BlobAccessTier,
-		}
+		return &InfoBlob{Blob: blobClient}, nil
 	}
-	return fileBlob, nil
+	return &BlockBlob{
+		Blob:           blobClient,
+		Indexes:        []int{},
+		BlobAccessTier: service.BlobAccessTier,
+	}, nil
 }
 
 // Delete the blockBlob from Azure Blob Storage
 func (blockBlob *BlockBlob) Delete(ctx context.Context) error {
-	_, err := blockBlob.Blob.Delete(ctx, azblob.DeleteSnapshotsOptionInclude, azblob.BlobAccessConditions{})
+	_, err := blockBlob.Blob.Delete(ctx, nil)
```

Comment:
Should we keep `DeleteSnapshotsOptionInclude` or whatever its newer form is?

Reply:
yes, reintroduced that